plugins/@grouparoo/clickhouse/src/lib/export/exportRecord.ts

```diff
@@ -0,0 +1,130 @@
+
+  table = table?.toString();
+  primaryKey = primaryKey?.toString();
+  groupsTable = groupsTable?.toString();
+  groupForeignKey = groupForeignKey?.toString();
+  groupColumnName = groupColumnName?.toString();
+
+  if (Object.keys(newRecordProperties).length === 0) {
+    return { success: true };
+  }
+
+  if (!newRecordProperties[primaryKey]) {
+    throw new Error(
+      `newRecordProperties[primaryKey] (${primaryKey}) is a required mapping`
+    );
+  }
+
+  try {
+    // --- Records --- //
+    if (toDelete) {
+      await connection.asyncQuery(`ALTER TABLE ?? DELETE WHERE ?? = ?`, [
+        table,
+        primaryKey,
+        newRecordProperties[primaryKey],
+      ]);
+    } else if (newRecordProperties[primaryKey]) {
+      const existingRecords = await connection.asyncQuery(
+        `SELECT * FROM ?? WHERE ?? = ?`,
+        [table, primaryKey, newRecordProperties[primaryKey]]
+      );
+
+      if (existingRecords.length === 1) {
+        const { [primaryKey]: _, ...columnsToUpdate } = newRecordProperties;
+
+        // update
+        await connection.asyncQuery(`ALTER TABLE ?? UPDATE ? WHERE ?? = ?`, [
+          table,
+          columnsToUpdate,
+          primaryKey,
+          newRecordProperties[primaryKey],
+        ]);
+
+        // erase old columns
+        const columnsToErase = Object.keys(existingRecords[0]).filter(
+          (k) =>
+            k !== primaryKey &&
+            (newRecordProperties[k] === null ||
+              newRecordProperties[k] === undefined) &&
+            (oldRecordProperties[k] !== null ||
+              oldRecordProperties[k] !== undefined)
+        );
+
+        if (columnsToErase.length > 0) {
+          const nullData = {};
+          columnsToErase.forEach((k) => (nullData[k] = null));
+
+          await connection.asyncQuery(`ALTER TABLE ?? UPDATE ? WHERE ?? = ?`, [
+            table,
+            nullData,
+            primaryKey,
+            newRecordProperties[primaryKey],
+          ]);
+        }
+      } else {
+        // delete & insert
+        await connection.asyncQuery(`ALTER TABLE ?? DELETE WHERE ?? = ?`, [
+          table,
+          primaryKey,
+          newRecordProperties[primaryKey],
+        ]);
+
+        await connection.asyncQuery(
+          `INSERT INTO ?? (${Object.keys(newRecordProperties).join(
+            ", "
+          )}) FORMAT Values (?)`,
+          [table, Object.values(newRecordProperties)]
+        );
+      }
+    } else {
+      // just insert
+      await connection.asyncQuery(
+        `INSERT INTO ?? (${Object.keys(newRecordProperties).join(
+          ", "
+        )}) FORMAT Values (?)`,
+        [table, Object.values(newRecordProperties)]
+      );
+    }
+
+    // --- Groups --- //
+
+    await connection.asyncQuery(`ALTER TABLE ?? DELETE WHERE ?? = ?`, [
+      groupsTable,
+      groupForeignKey,
+      newRecordProperties[primaryKey],
+    ]);
+
+    // TODO: Do we need this in ClickHouse? How do we translate this to ClickHouse?
```

Comment:
There's a slower pattern we could use where we do 2 queries: 
1. delete all the groups that exist for the record (`delete from groups where user_id = x`)
2. add the groups again